src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectDependencyProvider.cs

```diff
@@ -0,0 +1,158 @@
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Execution;
+using Microsoft.Extensions.ProjectModel.Resolution;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class MsBuildProjectDependencyProvider
+    {
+        private const string PackageDependencyItemType = "_DependenciesDesignTime";
+        private const string ResolvedReferenceItemType = "ReferencePath";
+
+        private readonly ProjectInstance _projectInstance;
+        public MsBuildProjectDependencyProvider(ProjectInstance projectInstance)
+        {
+            if (projectInstance == null)
+            {
+                throw new ArgumentNullException(nameof(projectInstance));
+            }
+            _projectInstance = projectInstance;
+        }
+
+        public IEnumerable<DependencyDescription> GetPackageDependencies()
+        {
+            var packageItems = _projectInstance.GetItems(PackageDependencyItemType);
+            var packageInfo = new Dictionary<string, DependencyDescription>(StringComparer.OrdinalIgnoreCase);
+            if (packageItems != null)
+            {
+                foreach (var packageItem in packageItems)
+                {
+                    var packageDependency = CreateDependencyDescriptionFromItem(packageItem);
+                    if (packageDependency != null)
+                    {
+                        packageInfo[packageItem.EvaluatedInclude] = packageDependency;
+                    }
+                }
+
+                // 2nd pass to populate dependencies;
+
+                PopulateDependencies(packageInfo, packageItems);
+            }
+
+            return packageInfo.Values;
+        }
+
+
+        public IEnumerable<ResolvedReference> GetResolvedReferences()
+        {
+            var refItems = _projectInstance.GetItems(ResolvedReferenceItemType);
+            var resolvedReferences = new List<ResolvedReference>();
+            if (refItems != null)
+            {
+                foreach (var refItem in refItems)
+                {
+                    var resolvedReference = CreateResolvedReferenceFromProjectItem(refItem);
+                    resolvedReferences.Add(resolvedReference);
+                }
+            }
+
+            return resolvedReferences;
+        }
+
+        private static ResolvedReference CreateResolvedReferenceFromProjectItem(ProjectItemInstance item)
+        {
+            const string ItemSpecKey = "OriginalItemSpec";
+            const string NugetSourceTypeKey = "NugetSourceType";
+            const string VersionKey = "Version";
+
+
+            string itemSpec = string.Empty;
+            string nugetPackageSource = string.Empty;
+            string version = string.Empty;
+            string resolvedPath = string.Empty;
+            string name = string.Empty;
+
+            resolvedPath = item.EvaluatedInclude;
+
+            name = string.IsNullOrEmpty(resolvedPath) ? "Unknown" : Path.GetFileNameWithoutExtension(resolvedPath);
```

Comment:
Good point. I think, it would be more justifiable to just skip this item if we don't have a resolved path. 
Coz, if we don't have resolved path, the only thing we would have is Version.